packages/app/src/components/DynamicRoot/DynamicRoot.tsx

```diff
@@ -0,0 +1,195 @@
+        AppRouter: React.ComponentType;
+        dynamicRoutes: DynamicRootContextValue[];
+        mountPoints: { [mountPoint: string]: ScalprumMountPoint[] };
+      }
+    | undefined
+  >();
+  const { initialized, pluginStore } = useScalprum();
+  // console.log({ DEFAULT_PROXY_PATH })
+
+  // Fills registry of remote components
+  const initializeRemoteModules = useCallback(async () => {
+    const { dynamicRoutes, mountPoints, routeBindings } =
+      extractDynamicConfig();
+
+    const requiredModules = [
+      ...mountPoints.map(({ module, scope }) => ({
+        scope,
+        module,
+      })),
+      ...dynamicRoutes.map(({ scope, module }) => ({
+        scope,
+        module,
+      })),
+    ];
+
+    const remotePlugins = await initializeRemotePlugins(
+      pluginStore,
+      scalprumConfig,
+      requiredModules,
+    );
+
+    if (!app.current) {
+      app.current = createApp({
+        apis,
+        bindRoutes({ bind }) {
+          bindAppRoutes(bind, remotePlugins, routeBindings);
+        },
+        themes: defaultThemes,
+        components: defaultAppComponents,
+      });
+    }
+
+    const providerMountPoints = mountPoints.map(
+      ({ module, importName, mountPoint, scope }) => ({
+        mountPoint,
+        Component: remotePlugins[scope][module][importName],
+      }),
+    );
+
+    const mountPointComponents = providerMountPoints.reduce<{
+      [mountPoint: string]: ScalprumMountPoint[];
+    }>((acc, entry) => {
+      if (!acc[entry.mountPoint]) {
+        acc[entry.mountPoint] = [];
+      }
+      acc[entry.mountPoint].push(entry.Component);
+      return acc;
+    }, {});
+    getScalprum().api.mountPoints = mountPointComponents;
+    const dynamicRoutesComponents = dynamicRoutes.map(route => ({
+      ...route,
+      Component: remotePlugins[route.scope][route.module][route.importName],
+    }));
+    setComponents({
+      AppProvider: app.current.getProvider(),
+      AppRouter: app.current.getRouter(),
+      dynamicRoutes: dynamicRoutesComponents,
+      mountPoints: mountPointComponents,
+    });
+    afterInit().then(({ default: Component }) => {
+      setChildComponent(() => Component);
+    });
+  }, [pluginStore, scalprumConfig, apis, afterInit]);
+
+  useEffect(() => {
+    if (initialized && !components) {
+      initializeRemoteModules();
+    }
+  }, [initialized, components, initializeRemoteModules]);
+
+  if (!initialized || !components) {
+    return null;
+  }
+
+  return (
+    <DynamicRootContext.Provider value={components}>
+      {ChildComponent ? <ChildComponent /> : <div>Loading</div>}
+    </DynamicRootContext.Provider>
+  );
+};
+
+const ScalprumRoot = ({
+  apis,
+  afterInit,
+}: {
+  // Static APIs
+  apis: AnyApiFactory[];
+  afterInit: () => Promise<{ default: React.ComponentType }>;
+}) => {
+  const config = useMemo(
+    () =>
+      ConfigReader.fromConfigs(
+        overrideBaseUrlConfigs(process.env.APP_CONFIG as any),
```

Comment:
This does not work, since it doesn't support runtime AppConfig injection.
See this commit for a hacky fix (which possibly can be made without duplicating the `defaultConfigLoaderSync()` function which is not exported (only the async `defaultConfigLoaderSync()` is exported, but I didn't know how to use it inside the JSX Component code) :
https://github.com/janus-idp/backstage-showcase/pull/682/commits/0b10be8ebb45f89924dd2595e5b82364acff4bf4

Reply:
cc @tumido